packages/colorpickers/src/elements/ColorPicker/ColorWell.tsx

```diff
@@ -22,37 +22,52 @@ interface IColorWellProps {
   hue: number;
   saturation: number;
   lightness: number;
-  onChange?: (hsv: IHSVColor, event: React.MouseEvent<any>) => void;
+  onChange?: (hsv: IHSVColor, event: MouseEvent) => void;
 }
 
 export const ColorWell: React.FC<IColorWellProps> = ({ hue, saturation, lightness, onChange }) => {
   const { rtl } = useContext(ThemeContext);
   const container = useRef<HTMLDivElement>(null);
   const hsv = hsl2hsv(hue, saturation, lightness);
 
-  const throttledChange = throttle(e => {
-    if (container.current) {
-      const nextHsv = calculateNextHsv(e, hsv, container.current, rtl);
+  const throttledChange = useMemo(() => {
+    /* eslint-disable-next-line @typescript-eslint/no-shadow */
+    return throttle((e: MouseEvent, hsv: IHSVColor, rtl: boolean) => {
+      if (container.current) {
+        const nextHsv = calculateNextHsv(e, hsv, container.current, rtl);
 
-      onChange && onChange(nextHsv, e);
-    }
-  }, 50);
+        onChange && onChange(nextHsv, e);
+      }
+    }, 50);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
-  const topPosition = 100 - hsv.v;
-  const leftPosition = rtl ? 100 - hsv.s : hsv.s;
+  const handleMouseMove = (e: MouseEvent) => throttledChange(e, hsv, rtl);
 
   const handleMouseUp = () => {
     throttledChange.cancel();
-    window.removeEventListener('mousemove', throttledChange);
+    window.removeEventListener('mousemove', handleMouseMove);
     window.removeEventListener('mouseup', handleMouseUp);
   };
 
   const handleMouseDown = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
-    throttledChange(e);
-    window.addEventListener('mousemove', throttledChange);
+    throttledChange(e as any, hsv, rtl);
+    window.addEventListener('mousemove', handleMouseMove);
     window.addEventListener('mouseup', handleMouseUp);
   };
 
+  useEffect(() => {
+    return () => {
+      throttledChange.cancel();
+      window.removeEventListener('mousemove', handleMouseMove);
+      window.removeEventListener('mouseup', handleMouseUp);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [throttledChange]);
```

Comment:
Is there a reason that we don't include the `handleMouseMove` and `handleMouseUp` in the dependency array? I think the warnings here are valid. On each render these callbacks would receive new references and not be removed correctly (except in the mouse up logic).

Some suggestions:
- Remove all of the rule-of-hooks exceptions
- Since `throttledChange` is now memorized we should migrate the mouse handlers to `useCallback`